code/go/0chain.net/smartcontract/dbs/event/snapshot.go

```diff
@@ -0,0 +1,201 @@
+package event
+
+import (
+	"0chain.net/chaincore/state"
+	"github.com/0chain/common/core/logging"
+	"gorm.io/gorm/clause"
+
+	"go.uber.org/zap"
+
+	"github.com/0chain/common/core/currency"
+)
+
+//max_capacity - maybe change it max capacity in blobber config and everywhere else to be less confusing.
+//staked - staked capacity by delegates
+//unstaked - opportunity for delegates to stake until max capacity
+//allocated - clients have locked up storage by purchasing allocations
+//unallocated - this is equal to (staked - allocated) and allows clients to purchase allocations with free space blobbers.
+//used - this is the actual usage or data that is in the server.
+//staked + unstaked = max_capacity
+//allocated + unallocated = staked
+
+// swagger:model Snapshot
+type Snapshot struct {
+	Round int64 `gorm:"primaryKey;autoIncrement:false" json:"round"`
+
+	TotalMint            int64 `json:"total_mint"`
```

Comment:
Please use currency.Coin for tokens, and use currency package functions for tokens calculation. Same as other fields below

Reply:
@peterlimg my main concern about currency here that it's uint64 and sometimes aggregates can be negative, it's not coins, but some sort of derivative, so I left it as ints which are more like points, but not real tokens.